nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -40,33 +42,38 @@ let
       empty_file=$(${pkgs.coreutils}/bin/mktemp)
 
       ${concatStrings (mapAttrsToList (n: v: ''
-        ${pkgs.coreutils}/bin/install -Dp "${v}" "${efi.efiSysMountPoint}/"${escapeShellArg n}
-        ${pkgs.coreutils}/bin/install -D $empty_file "${efi.efiSysMountPoint}/efi/nixos/.extra-files/"${escapeShellArg n}
+        ${pkgs.coreutils}/bin/install -Dp "${v}" "${mountPoint}/"${escapeShellArg n}
+        ${pkgs.coreutils}/bin/install -D $empty_file "${mountPoint}/efi/nixos/.extra-files/"${escapeShellArg n}
       '') cfg.extraFiles)}
 
       ${concatStrings (mapAttrsToList (n: v: ''
-        ${pkgs.coreutils}/bin/install -Dp "${pkgs.writeText n v}" "${efi.efiSysMountPoint}/loader/entries/"${escapeShellArg n}
-        ${pkgs.coreutils}/bin/install -D $empty_file "${efi.efiSysMountPoint}/efi/nixos/.extra-files/loader/entries/"${escapeShellArg n}
+        ${pkgs.coreutils}/bin/install -Dp "${pkgs.writeText n v}" "${mountPoint}/loader/entries/"${escapeShellArg n}
+        ${pkgs.coreutils}/bin/install -D $empty_file "${mountPoint}/efi/nixos/.extra-files/loader/entries/"${escapeShellArg n}
       '') cfg.extraEntries)}
     '';
   };
 
-  checkedSystemdBootBuilder = pkgs.runCommand "systemd-boot" {
+  checkedSystemdBootBuilder = mountPoint: pkgs.runCommand "systemd-boot" {
     nativeBuildInputs = [ pkgs.mypy python3 ];
   } ''
-    install -m755 ${systemdBootBuilder} $out
+    install -m755 ${systemdBootBuilder mountPoint} $out
     mypy \
       --no-implicit-optional \
       --disallow-untyped-calls \
       --disallow-untyped-defs \
       $out
   '';
 
-  finalSystemdBootBuilder = pkgs.writeScript "install-systemd-boot.sh" ''
-    #!${pkgs.runtimeShell}
-    ${checkedSystemdBootBuilder} "$@"
-    ${cfg.extraInstallCommands}
-  '';
+  finalSystemdBootBuilder = let
+    installDirs =
+      if cfg.mirroredBoots != []
+      then cfg.mirroredBoots
+      else [efi.efiSysMountPoint];
```

Comment:
My first thought was that mirroredBoots should be *additional* mountpoints, but seeing this I guess the implementation is *only these* mountpoints (the "main" ESP mountpoint gets ignored)? Am I the only one that find that surprising?

Reply:
Indeed, and AFAIU the `grub.nix` case, `mirroredBoots` are *additionals* mountpoints there: https://github.com/NixOS/nixpkgs/blob/ec6cc9c826e08ec1aa4212c40ad92eb5186a9053/nixos/modules/system/boot/loader/grub/grub.nix#L708-L710